pkg/cloudprovider/aws/amifamily/resolver.go

```diff
@@ -96,13 +97,9 @@ func (r Resolver) Resolve(ctx context.Context, provider *v1alpha1.AWS, nodeReque
 		return nil, err
 	}
 	amiFamily := GetAMIFamily(provider.AMIFamily, options)
-	amiIDs := map[string][]cloudprovider.InstanceType{}
-	for _, instanceType := range nodeRequest.InstanceTypeOptions {
-		amiID, err := r.amiProvider.Get(ctx, instanceType, amiFamily.SSMAlias(options.KubernetesVersion, instanceType))
-		if err != nil {
-			return nil, err
-		}
-		amiIDs[amiID] = append(amiIDs[amiID], instanceType)
+	amiIDs, err := r.amiProvider.Get(ctx, provider, nodeRequest, options, amiFamily)
```

Comment:
I think it's going to be required to do an inspection of the AMI to see what the block device mappings are and potentially other metadata that we'll need for custom AMIs. It's probably better to do that anyways for the default AMIs too. The default AMIs could potentially be updated to use different block device mappings that we should really discover.

Reply:
> The default AMIs could potentially be updated to use different block device mappings that we should really discover.


I know AMIs are versioned so backwards compatibility is less of a concern, but even so I don't expect something like the default device name / other metadata to change for a given AMI family. Can break everyone's existing launch templates pretty easily :D

> I think it's going to be required to do an inspection of the AMI to see what the block device mappings are and potentially other metadata that we'll need for custom AMIs

The thinking I had on this was that when you specify the AMIFamily to be `AL2` or `Bottlerocket` then we'll just assume the same settings as that AMIFamily. If you want to override other metadata and things, then you have to specify `AMIFamily` as `Custom` and then we'll start inspecting the AMI and get the `RootDeviceName` from there. It's a fair amount of changes to support that AMIFamily (new IAM permissions for ec2:DescribeImages too) which is why I thought it deserved it's own PR. 

Thoughts?

Reply:
👍 sgtm on an additional PR if it becomes necessary. The block device mappings I was thinking of that could change is Ubuntu. Since we pick up latest, it might break unintentionally. But I'm not overly concerned.